### DISCUSSION TITLE
baseapp崩溃, 通过core文件定位在onLoseCell

### DISCUSSION THREAD

**Discussion by miaoshujiang** [kbengine/kbengine#1531] [Q&A]
![image](https://user-images.githubusercontent.com/23332574/148191278-d46aa5eb-92f5-45fc-bc6d-732ae9f819b0.png)

**Comment by miaoshujiang** [comment]
@kebiao

**Comment by kbengine** [comment]
检查一下脚本onlosecell中是不是又调用了destroy之类的



发自我的iPhone


------------------ 原始邮件 ------------------
发件人: miaoshujiang ***@***.***&gt;
发送时间: 2022年1月5日 18:39
收件人: kbengine/kbengine ***@***.***&gt;
抄送: Subscribed ***@***.***&gt;
主题: Re: [kbengine/kbengine] baseapp崩溃, 通过core文件定位在onLoseCell (Discussion #1531)





 
@kebiao
 
—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Reply by miaoshujiang** [reply]
onLoseCell 调用destroy不是挺正常的吗？ 玩家下线后, 触发onClientDeath, 等到cell销毁后然后destroy玩家

**Comment by kbengine** [comment]
gdb bt命令看看，多发一点信息。 日志贴一下



发自我的iPhone


------------------ 原始邮件 ------------------
发件人: miaoshujiang ***@***.***&gt;
发送时间: 2022年1月5日 18:39
收件人: kbengine/kbengine ***@***.***&gt;
抄送: Subscribed ***@***.***&gt;
主题: Re: [kbengine/kbengine] baseapp崩溃, 通过core文件定位在onLoseCell (Discussion #1531)





 
@kebiao
 
—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Reply by miaoshujiang** [reply]
![image](https://user-images.githubusercontent.com/23332574/148479922-4407eb62-d5b5-4651-b80a-fdf6ed3d3f10.png)

**Reply by miaoshujiang** [reply]
![image](https://user-images.githubusercontent.com/23332574/148482202-4ba341fa-c812-4e28-a003-048f63696044.png)

**Reply by huhubis** [reply]
能说一下具体是怎么产生的吗？ 我看你这里是个握手协议触发的。  有可重现脚本吗？打个包发上来看看， 最好是服务器跑起来，运行bots就能触发

**Reply by miaoshujiang** [reply]
> 能说一下具体是怎么产生的吗？ 我看你这里是个握手协议触发的。 有可重现脚本吗？打个包发上来看看， 最好是服务器跑起来，运行bots就能触发

暂时重现不了, 是大量线上玩家很难才触发一次, 但是好像只要一触发每次都是在这里引发

**Reply by huhubis** [reply]
是原版吗？ core文件发上来？

**Reply by miaoshujiang** [reply]
[core.zip](https://github.com/kbengine/kbengine/files/7827146/core.zip)
1.3.7 版本 修改过源码, baseapp暂时传不上来,太大

**Comment by huhubis** [comment]
你版本太老了，改过源码就不知道了。
你这个看起来和fmt格式化有关。
你看看这个提交有没有解决你的问题
![image](https://user-images.githubusercontent.com/80089038/148525806-92c8add3-79ef-429f-84fd-b4dd027ef92a.png)

**Reply by miaoshujiang** [reply]
![image](https://user-images.githubusercontent.com/23332574/148629890-38b58b18-82d7-4a19-be99-1aae716813cc.png)
均定位到此断开
